nasa-gibs/worldview
# WV-2285: Don't allow shortening URLs that are too long

## Description

Fixes # .

A URL over 2048 characters cannot be shortened via the link shortener. I have added code to check the length of the current URL & suppress the Shorten Link component if the URL length exceeds 2048.

## How To Test

To test you need to exceeds 2048 characters in the URL. To do so, load a fresh instance & select "Add Layers" at the bottom of the window on the left side. If you enable every layer option in Carbon Monoxide, Corrected Reflectance, Dust, and then just 1 option in Fire & Thermal anomalies the URL will exceed 2048. (You can check the URL length as you add layers with window.location.href.length in the console) Once you have exceeded 2048, close the layers window & click the "Share" option on the top-right of the screen. "Shorten URL" & the checkbox will not be rendered.

Now, you can delete any layer (click the 'x' on any item in the left window) & the 'Shorten Link" option will render once the URL is again below or equal to 2048 characters.

You can re-add layers to confirm that the rendering is behaving appropriately as the URL length changes.
